From a11f11fe4bf5ef3d3359033e46f3e442a75dd9c8 Mon Sep 17 00:00:00 2001
From: Timur Tabi <timur@freescale.com>
Date: Tue, 15 Sep 2009 17:16:40 -0500
Subject: [PATCH 038/148] p4080ds: implement work-around for erratum CPU6

Do not enable interrupts right before the fourth (and final) call to tlbsync.

Erratum CPU6 says that ME, CE, DE, and EE cannot be enabled when executing a
tlbsync.  The kernel uses tlbsync four times during early boot-up.

A recent change in U-Boot ("ppc/85xx: Disable all async interrupt
sources when we boot") disabled all of these when the kernel boots, but
the kernel turns them back on right before the final call to tlbsync.
However, interrupts dont' need to be enabled at this point; and indeed,
the kernel redundantly enables them again a short time later.  So we
change the code not to enable interrupts right before that last tlbsync
instruction.

Signed-off-by: Timur Tabi <timur@freescale.com>
[Cleanly applied the FSL SDK 2.0.3 patch:
"kernel-2.6.30-p4080ds-implement-work-around-for-erratum-CP.patch"]
Integrated-by: Yuri Nedel <Yuri.Nedel@windriver.com>
---
 arch/powerpc/kernel/head_fsl_booke.S |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/arch/powerpc/kernel/head_fsl_booke.S b/arch/powerpc/kernel/head_fsl_booke.S
index 7acbc8a..034cfd8 100644
--- a/arch/powerpc/kernel/head_fsl_booke.S
+++ b/arch/powerpc/kernel/head_fsl_booke.S
@@ -275,8 +275,7 @@ skpinv:	addi	r6,r6,1				/* Increment */
 /* 7. Jump to KERNELBASE mapping */
 	lis	r6,(KERNELBASE & ~0xfff)@h
 	ori	r6,r6,(KERNELBASE & ~0xfff)@l
-	lis	r7,MSR_KERNEL@h
-	ori	r7,r7,MSR_KERNEL@l
+	li	r7,0
 	bl	1f			/* Find our address */
 1:	mflr	r9
 	rlwimi	r6,r9,0,20,31
-- 
1.6.5.2

